### PR TITLE
Homepage boards and projects

### PR DESCRIPTION
**Pull request by ppivanov**
Fixed Entity Framework mapping issues (foreign key stuff I've missed while refactoring the schema).

Created 4 new methods on User that retrieve boards or projects where the user has access.
1. GetBoardsWhereThisUserIsOwner();
2. GetBoardsWhereThisUserIsMember();
3. GetProjectsWhereThisUserIsOwner();
4. GetProjectsWhereThisUserIsMember();

Created a new class 'HomePageModel' that will hold all the information to be displayed on the home page.
Changed the [Post] Login method on AccountController so that, whenever login is successful, to instantiate a new HomePageModel and redirect to Index.cshtml.

Modified Index.cshtml so it displays the information from the incoming object.

### CODE REVIEW

**Review by HelpMichaelCode — APPROVED** [review]

**Review by HelpMichaelCode — APPROVED** [review]

File: BoardWebApp-EF/Controllers/AccountController.cs [hunk 1]
```diff
@@ -54,9 +54,11 @@ public IActionResult LogoutSuccess(SendLoginModel msgObj)
         }
 
         [HttpGet]
-        public IActionResult Login()
+        public IActionResult Login(string errorMessage)
         {
-            return View();
+            SendLoginModel errorMessageModel = new SendLoginModel();
+            errorMessageModel.ValidationErrorMessages.Add(errorMessage);
```
Comment (HelpMichaelCode) [inline BoardWebApp-EF/Controllers/AccountController.cs:60]: Have a single string whenever a user enters in the wrong credentials.

File: BoardWebApp-EF/Controllers/HomeController.cs [hunk 2]
```diff
@@ -20,24 +22,30 @@ public HomeController(ILogger<HomeController> logger, BoardWebAppContext dbConte
         }
 
 #nullable enable
-        public IActionResult Index(string? errorMessage)
+        public IActionResult Index(HomePageModel homePageData)
         {
-            if (errorMessage is null)
+            if (homePageData.Message is null)
             {
```
Comment (HelpMichaelCode) [inline BoardWebApp-EF/Controllers/HomeController.cs:28]: Board and project messages not showing when user logs in

File: BoardWebApp-EF/Controllers/HomeController.cs [hunk 3]
```diff
@@ -20,24 +22,30 @@ public HomeController(ILogger<HomeController> logger, BoardWebAppContext dbConte
...
                 }
                 else
                 {
-                    return View((object)"BoardWebApp");
+                    Console.WriteLine("User IS authenticated");
+                    homePageData = new HomePageModel(authenticatedUser, _dbContext);
+                    return View(homePageData);
                 }
             }
             else
```
Comment (HelpMichaelCode) [inline BoardWebApp-EF/Controllers/HomeController.cs:45]: Remove this else

File: BoardWebApp-EF/Controllers/HomeController.cs [hunk 4]
```diff
@@ -20,24 +22,30 @@ public HomeController(ILogger<HomeController> logger, BoardWebAppContext dbConte
         }
 
 #nullable enable
-        public IActionResult Index(string? errorMessage)
+        public IActionResult Index(HomePageModel homePageData)
         {
-            if (errorMessage is null)
+            if (homePageData.Message is null)
```
Comment (HelpMichaelCode) [inline BoardWebApp-EF/Controllers/HomeController.cs:27]: Remove this if

File: BoardWebApp-EF/Models/Project.cs [hunk 5]
```diff
@@ -15,7 +16,9 @@ public Project()
         public string ProjectName { get; set; }
         public string ProjectDescription { get; set; }
 
+        [InverseProperty("Project")]
         public virtual ICollection<Board> Board { get; set; }
+        [InverseProperty("Project")]
         public virtual ICollection<ProjectMember> ProjectMember { get; set; }
```
Comment (HelpMichaelCode) [inline BoardWebApp-EF/Models/Project.cs:22]: Confused on how virtual works here with regards to foreign keys

**Review by HelpMichaelCode — COMMENTED** [review]
I am not approving this...

File: BoardWebApp-EF/Models/User.cs [hunk 6]
```diff
@@ -71,6 +72,54 @@ public static string ComputeSha256HashForString(string rawString)
...
             } 
         }
-                
+
+
+        //  Member Type   - Id
+        //  Owner         - 1
+        //  Member        - 2
+        //  Scrum Master  - 3
+
```
Comment (ppivanov) [inline BoardWebApp-EF/Models/User.cs:81]: Try and extract the query code in a separate method that takes in the member type as parameters.

**Pull request merged**